gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -707,16 +707,25 @@ private final PreparedStatement prepMinion(
 						qt.bindStatement();
 					}
 					catch(StandardException | AssertFailure ex) {
-				      boolean routeToSnappy = (ex instanceof StandardException && ((StandardException)ex).
-									getMessageId().equals(SQLState.ROW_LEVEL_SECURITY_ENABLED)) ||
-									(routeQuery && !NON_ROUTED_QUERY.matcher(source).find());
+						  boolean isDML = DML_TABLE_PATTERN.matcher(source).find();
+						  String messageId = ex instanceof StandardException ? ((StandardException)ex).getMessageId() : "";
+						  boolean routeToSnappy = messageId.equals(SQLState.ROW_LEVEL_SECURITY_ENABLED) ||
+									(routeQuery && !NON_ROUTED_QUERY.matcher(source).find()
+									//SNAP-2765 don't route if failure due to error such as invalid column name in prepared statement
+									&& !(isDML && this.isPreparedStatement() &&
+											(messageId.equals(SQLState.LANG_COLUMN_NOT_FOUND_IN_TABLE)
+											|| messageId.equals(SQLState.LANG_DB2_INVALID_COLS_SPECIFIED)
```

Comment:
Cosmetic change suggestion. We can just move these conditions in a private boolean method so that this code block does not seem bloated.

Reply:
Done